File: COOLSOUND-FRONTEND/coolsound/src/app/services/playlist.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Playlist } from '../model/playlist';

@Injectable({
  providedIn: 'root'
})
export class PlaylistService {

  private baseUrl = '/playlist/';

  constructor(private http: HttpClient) { }

  /** Método para sacar las canciones de BD */
  playlist(): Observable<any> {
    return this.http.get<Playlist>(`${this.baseUrl}` + 'getPlaylists/');
  }

  /** Método para añadir Comentarios a BD */
  nuevaPlaylist(texto: string, idUsuario: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'nuevaPlaylist/' + texto + '/' + idUsuario);
  }

  /** Método para eliminar Playlists de BD */
  eliminarPlaylist(idPlaylist: number): Observable<any> {
    return this.http.get(`${this.baseUrl}` + 'eliminarPlaylist/' + idPlaylist);
  }
}

File: COOLSOUND-BACKEND/coolsound/src/main/java/com/java/coolsound/service/PlaylistService.java
package com.java.coolsound.service;

import java.util.List;

import com.java.coolsound.model.Playlist;

public interface PlaylistService {

	/**
	 * Gets the all Playlists.
	 *
	 * @return the all playlists
	 */
	List<Playlist> getAllPlaylists();
	
	/**
	 * Gets the all Playlists by IdCancion.
	 *
	 * @return the all playlists
	 */
	List<Playlist> getAllPlaylistsById(int idUsuario);
	
	/**
	 * Añadir Playlists.
	 *
	 * @return añadir Playlists
	 */
	int nuevaPlaylist(String nombrePlaylist, int idUsuario);
	
	/**
	 * Eliminar Playlists.
	 *
	 * @return eliminar Playlists
	 */
	int eliminarPlaylist(int idPlaylist);
	
}
